tests/ts/examples.ts

```diff
@@ -584,6 +593,107 @@ Person.query().where(ref('Model.jsonColumn:details'), '=', lit({ name: 'Jennifer
+const findByIdJoinRawThrow: Promise<Person> = Person.query()
+  .findById(32)
+  .joinRaw('raw sql')
+  .throwIfNotFound();
+const findOneWhere: Promise<Person | undefined> = Person.query()
+  .findOne('raw sql')
+  .where('more raw sql');
+const findOneWhereThrow: Promise<Person> = Person.query()
+  .findOne('raw sql')
+  .where('more raw sql')
+  .throwIfNotFound();
+const findOneSelect: Promise<Person | undefined> = Person.query()
+  .findOne('raw sql')
+  .select('firstName');
+const findOneSelectThrow: Promise<Person> = Person.query()
+  .findOne('raw sql')
+  .select('firstName')
+  .throwIfNotFound();
+const findOneWhereIn: Promise<Person | undefined> = Person.query()
+  .findOne('raw sql')
+  .whereIn('status', ['active', 'pending']);
+const findOneWhereInThrow: Promise<Person> = Person.query()
+  .findOne('raw sql')
+  .whereIn('status', ['active', 'pending'])
+  .throwIfNotFound();
+const findOneWhereJson: Promise<Person | undefined> = Person.query()
+  .findOne('raw sql')
+  .whereJsonSupersetOf('x:y', 'abc');
+const findOneWhereJsonThrow: Promise<Person> = Person.query()
+  .findOne('raw sql')
+  .whereJsonSupersetOf('x:y', 'abc')
+  .throwIfNotFound();
+const findOneWhereJsonIsArray: Promise<Person | undefined> = Person.query()
+  .findOne('raw sql')
+  .whereJsonIsArray('x:y');
+const findOneWhereJsonIsArrayThrow: Promise<Person> = Person.query()
+  .findOne('raw sql')
+  .whereJsonIsArray('x:y')
+  .throwIfNotFound();
+const patchWhere: Promise<number> = Person.query()
+  .patch({ firstName: 'Mo' })
+  .where('id', 32);
+const patchWhereIn: Promise<number> = Person.query()
+  .patch({ firstName: 'Mo' })
+  .whereIn('id', [1, 2, 3]);
+const patchWhereJson: Promise<number> = Person.query()
+  .patch({ firstName: 'Mo' })
+  .whereJsonSupersetOf('x:y', 'abc');
+const patchWhereJsonIsArray: Promise<number> = Person.query()
+  .patch({ firstName: 'Mo' })
+  .whereJsonIsArray('x:y');
+const patchThrow: Promise<number> = Person.query()
+  .patch({ firstName: 'Mo' })
+  .throwIfNotFound();
+const updateWhere: Promise<number> = Person.query()
+  .update({ firstName: 'Mo' })
+  .where('id', 32);
+const updateWhereIn: Promise<number> = Person.query()
+  .update({ firstName: 'Mo' })
+  .whereIn('id', [1, 2, 3]);
+const updateWhereJson: Promise<number> = Person.query()
+  .update({ firstName: 'Mo' })
+  .whereJsonSupersetOf('x:y', 'abc');
+const updateWhereJsonIsArray: Promise<number> = Person.query()
+  .update({ firstName: 'Mo' })
+  .whereJsonIsArray('x:y');
+const updateThrow: Promise<number> = Person.query()
+  .update({ firstName: 'Mo' })
+  .throwIfNotFound();
+const deleteWhere: Promise<number> = Person.query()
+  .delete()
+  .where('raw sql');
+const deleteWhereIn: Promise<number> = Person.query()
+  .delete()
+  .whereIn('id', [1, 2, 3]);
+const deleteThrow: Promise<number> = Person.query()
+  .delete()
+  .throwIfNotFound();
+const deleteByIDThrow: Promise<number> = Person.query()
+  .deleteById(32)
+  .throwIfNotFound();
+
```

Comment:
Are there tests for these cases:

```js
// first turns array result into single
const x1: Promise<Model> = qb.where().first();
// the location of `first` doesn't matter
const x2: Promise<Model> = qb.first().where();
// returning turns `number` result back to `M[]`
const x3: Promise<Model[]> = qb.where().patch().returning();
// the position of `returning` doesn't matter (as long as it comes after the method that turns type into non-model)
const x4: Promise<Model> = qb.delete().returning().first().where();
```

Reply:
I'll check and add them if not. I want more `returning()` test cases as well. We put `returning()` on the base query builder, so it's always accessible.

Reply:
I'm noticing that my new examples are only testing the static methods. I wonder if I should duplicate them all for the instance methods.

Reply:
I'd tried to hit both in the past. If you could add them, that'd be great.

Reply:
Maybe we don't need to dup many tests. After the first call to `Model.query()` or `model.$query()`, we're testing the same query builder methods. I'm thinking that we can provide coverage either way and then throw in a few token tests for other methods of getting query builders. The only internals we're taking advantage of is our knowing that they're all using the same query builder types.

Less code duplication makes for easier maintenance. And less judging! What do you think?